### PR TITLE
ARROW-15824: [Python] Make pyarrow.parquet a package

### PR DESCRIPTION
**Pull request by raulcd**
This PR makes `pyarrow.parquet` a package and moves `pyarrow.parquet_encryption` to `pyarrow.parquet.encryption`

### CODE REVIEW

**Comment by raulcd** [comment]
@jorisvandenbossche I've opted to just move the previous `parquet.py` to the `parquet/__init__.py` package file. Do you think we should also split the implementation from the `__init__.py` to its own implementation file (like `parquet/parquet.py`) and only expose the API on `__init__.py`?
From my understanding this is the minimal change for the requested ticket.

**Comment by AlenkaF** [comment]
Thanks for this, looks good to me 👍

**Review by andersonm-ibm — CHANGES_REQUESTED** [review]
Thank you for these changes!

File: docs/source/python/parquet.rst [hunk 1]
```diff
@@ -683,7 +683,7 @@ For example, in order to use the ``MyKmsClient`` defined above:
 
    crypto_factory = CryptoFactory(kms_client_factory)
 
-An :download:`example <../../../python/examples/parquet_encryption/sample_vault_kms_client.py>`
+An :download:`example <../../../python/examples/parquet.encryption/sample_vault_kms_client.py>`
```
[the remark this answers is not in the document]
Comment (raulcd) [reply docs/source/python/parquet.rst]: Thanks for the comment, you are correct, I have reverted back to the current examples directory: `python/examples/parquet_encryption/sample_vault_kms_client.py`

**Comment by wesm** [comment]
I apologize for the bikeshed moment, but I was debugging something and found that much of pyarrow/parquet.py had been relocated to pyarrow/parquet/__init__.py -- I have often believed that having non-trivial implementation code in __init__.py files was an antipattern (and we have some code in pyarrow/__init__.py and maybe elsewhere that might be better moved to a utility module). However, my beliefs may not be aligned with broader community practices in the Python community nowadays. Thoughts from other frequent Python contributors (@jorisvandenbossche, @pitrou)?

**Comment by jorisvandenbossche** [comment]
I certainly agree that it is indeed not a typical pattern to put so much code in the init file (it also makes it more difficult to open the file by name in an editor ..). As Raúl asked above, we could also move all this code into `pyarrow/parquet/parquet.py` (or some other name). 

I think one small advantage of the `__init__.py` in this case is that it leaves the path of the functions the same (`pyarrow.parquet.read_table`, instead of `pyarrow.parquet.parquet.read_table`), and whenever the actual path is not the public path, you will always get users that will start using full non-public path. But don't know if that's enough to say "practicality beats purity" in this case :)

**Comment by pitrou** [comment]
We could certainly move all code to `pyarrow/parquet/core.py` and just import the APIs from the `__init__.py`. Someone just needs to open a JIRA and submit a PR :-)

**Comment by raulcd** [comment]
I've created a ticket to track it https://issues.apache.org/jira/browse/ARROW-17106 and tagged as good-first-issue

**Pull request closed without merging**